frontend/src/lib/components/Popup/Popup.tsx

```diff
@@ -144,7 +152,15 @@ export const Popup = React.forwardRef<HTMLDivElement, PopupProps>(
 
         useOutsideClickHandler(
             [floatingRef, referenceRef, ...additionalRefs],
-            (event) => visible && onClickOutside?.(event),
+            (event) => {
+                // Delay by a tick to allow other Popups to detect inside clicks.
+                // If a nested popup has handled the click, don't do anything
+                setTimeout(() => {
+                    if (visible && !nestedPopupReceivedClick) {
+                        onClickOutside?.(event)
+                    }
+                }, 1)
```

Comment:
Technically `setTimeout(() => {}, 1)` might delay by more than a tick if the ticks are really fast.